frontend-utils/src/backends/audio.rs

```diff
@@ -1,12 +1,29 @@
-use crate::preferences::GlobalPreferences;
-use anyhow::{anyhow, Context, Error};
 use cpal::traits::{DeviceTrait, HostTrait, StreamTrait};
+use cpal::SampleFormat;
 use ruffle_core::backend::audio::{
     swf, AudioBackend, AudioMixer, DecodeError, RegisterError, SoundHandle, SoundInstanceHandle,
     SoundStreamInfo, SoundTransform,
 };
 use ruffle_core::impl_audio_mixer_backend;
 
+#[derive(Debug, thiserror::Error)]
+pub enum CpalError {
+    #[error("No audio devices available")]
+    NoDevices,
+
+    #[error("Failed to get default output config")]
+    DefaultStream(#[from] cpal::DefaultStreamConfigError),
+
+    #[error("Unsupported sample format {0:?}")]
+    UnsupportedSampleFormat(SampleFormat),
+
+    #[error("Couldn't play the audio stream")]
+    Play(#[from] cpal::PlayStreamError),
+
+    #[error("failed constructing audio stream")]
+    Build(#[from] cpal::BuildStreamError),
```

Comment:
nit: Capitalization - but also, is this actually used?

Reply:
it is, implicitly.

Reply:
Fixed the capitalization, and changed up the phrasing a little bit - I hope you don't mind!